src/QsFmt/Formatter/SyntaxTree/Namespace.fs

```diff
@@ -12,9 +12,11 @@ type TypeParameterBinding =
         CloseBracket: Terminal
     }
 
+type internal SpecializationParameter = Terminal
+
 type SpecializationGenerator =
     | BuiltIn of name: Terminal * semicolon: Terminal
-    | Provided of parameters: Terminal option * statements: Statement Block
+    | Provided of parameters: SpecializationParameter Tuple option * statements: Statement Block
```

Comment:
Since this is a Tuple type now, I don't think there needs to be an option type here.

Reply:
But the ANTLR rule is [`providedSpecialization : parameters=specializationParameterTuple? block=scope;`](https://github.com/microsoft/qsharp-compiler/blob/main/src/QsFmt/Parser/QSharpParser.g4#L110). Note the `?` after `specializationParameterTuple`. It means that the entire tuple can be missing, including the parentheses. I have also found some code examples that do not have `()`, e.g., `body { }`([CodegenTests.qs](https://github.com/microsoft/qsharp-compiler/blob/main/src/QsCompiler/Tests.CSharpGeneration/Circuits/CodegenTests.qs#L59)) and
[InternalRenaming.qs](https://github.com/microsoft/qsharp-compiler/blob/main/src/QsCompiler/Tests.Compiler/TestCases/LinkingTests/InternalRenaming.qs#L103):
```
        body {
        }
```
Maybe this style is also deprecated? In that case, we do not only need to remove the `option`, but also modify the parser and the examples. Maybe even add move update rules?

Reply:
Oh I didn't know we could just _not_ have the parameter tuple. That is also deprecated syntax, but to be able to handle it, we would need to keep the `option` type as you have it here. Could you add the mapping of `body { }` -> `body (...) { }` syntax to this rule, or is it a significant enough change that it needs to be a separate rule?

Reply:
I have added the supporting for `body { }` in recent commits. To add the `()`, I hard-coded an empty `Tuple` object,  which I hope is not a problem.

Reply:
Thanks!